marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/reader/ReadWithCustomEndpointsTest.java

```diff
@@ -0,0 +1,41 @@
+package com.marklogic.hub.spark.sql.sources.v2.reader;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.hub.spark.sql.sources.v2.Options;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ReadWithCustomEndpointsTest extends AbstractSparkReadTest {
+
+    @BeforeEach
+    void setup() {
+        installCustomEndpoint("custom-read-endpoints/initializeRead.api", "custom-read-endpoints/initializeRead.sjs");
+        installCustomEndpoint("custom-read-endpoints/customReadRows.api", "custom-read-endpoints/customReadRows.sjs");
+    }
+
+    @Test
+    void test() {
+        ObjectNode endpointState = objectMapper.createObjectNode();
+        ObjectNode testRow = endpointState.putObject("testRow");
+        testRow.put("myId", "12345");
+        testRow.put("myName", "This is a test row");
+
+        Options options = newOptions().withView("doesnt-matter")
+            .withInitializeReadApiPath("/custom-read-endpoints/initializeRead.api")
+            .withReadRowsApiPath("/custom-read-endpoints/customReadRows.api")
+            .withReadRowsEndpointState(endpointState);
+
+        List<InternalRow> rows = readRows(new HubDataSourceReader(options.toDataSourceOptions()));
```

Comment:
Once HubInputPartitionReader is updated to use the custom endpoint, this test should pass.